### PR TITLE
[libfork] update to version 3.8.0

### PR DESCRIPTION
**Pull request by ConorWilliams**
- [x]  Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by ConorWilliams** [comment]
Not sure why CI is failing, output is this:

```

Triplet: x64-osx
RESULTS
  vcpkg-cmake:x64-osx: SUCCEEDED: 27.6 ms
  vcpkg-cmake-config:x64-osx: SUCCEEDED: 11.9 ms
  vcpkg-tool-meson:x64-osx: SUCCEEDED: 19.4 ms
  pkgconf:x64-osx: SUCCEEDED: 38.2 ms
  hwloc:x64-osx: SUCCEEDED: 94.5 ms
  libfork:x64-osx: SUCCEEDED: 11 s

SUMMARY FOR x64-osx
  SUCCEEDED: 6

##[error]PowerShell wrote one or more lines to the standard error stream.
##[error]pwsh(10783) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.

##[error]pwsh(10786) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.

Finishing: *** Test Modified Ports
```

Looks like it should be green to me?

File: ports/libfork/vcpkg.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "libfork",
-  "version": "3.7.1",
+  "version": "3.8.0",
   "description": "A bleeding-edge, lock-free, wait-free, continuation-stealing fork-join library built on C++20's coroutines.",
   "homepage": "https://github.com/conorwilliams/libfork",
   "license": "MPL-2.0",
   "dependencies": [
-    "boost-atomic",
```
Comment (JonLiu1993) [inline ports/libfork/vcpkg.json:8]: @ConorWilliams, Thanks for your PR, could you please clarify why remove `boost-atomic` add `pkgconf` dependencies?

Comment (ConorWilliams) [reply ports/libfork/vcpkg.json:8]: libfork used to support ``boost-atomic`` as an optional dependency, this support has been dropped in 3.8 and replaced with a port of just a small portion that was used. The ``hwloc`` package exports a ``.pc`` file which requires ``pkgconf``` to find.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libfork/vcpkg.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
     "hwloc",
+    "pkgconf",
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/libfork/vcpkg.json:9]: BTW it should be verifies that the exported CMake config doesn't contain unresolved imported targets for hwloc.

[the remark this answers is not in the document]
Comment (ConorWilliams) [reply ports/libfork/vcpkg.json]: Thanks @dg0yt, I have made the changes you suggested

File: ports/libfork/portfile.cmake [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  OUT_SOURCE_PATH
+  SOURCE_PATH
+  REPO
+  conorwilliams/libfork
+  REF
+  "v${VERSION}"
+  SHA512
+  38a8a6fe0f360f1caa123b227996490f192f8b58340ecb5d91922c15d7ca9b364031716e139a3ab0d89cd7cdf3bfb22fcf75272e76a40513c55adaf00ff6454d
+  HEAD_REF
+  main)
```
Comment (JonLiu1993) [inline ports/libfork/portfile.cmake:11]: @ConorWilliams, Could you plesae restore the format of this portfile.cmake to its previous format? Comparative standards before.

**Review by JonLiu1993 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**